tests/jit_spec.rs

```diff
@@ -105,8 +124,72 @@ mod tests {
             }
         "#,
         );
-        let executor = new_executor(&request).await.unwrap();
-        let response = executor.execute(request).await;
+        let executor = TestExecutor::try_new().await.unwrap();
+        let response = executor.run(request).await.unwrap();
+        let data = response.data;
+
+        insta::assert_json_snapshot!(data);
+    }
+
+    #[tokio::test]
+    async fn test_executor_arguments() {
+        //  NOTE: This test makes a real HTTP call
+        let request = Request::new("query {user(id: 1) {id}}");
+        let executor = TestExecutor::try_new().await.unwrap();
+        let response = executor.run(request).await.unwrap();
+        let data = response.data;
+
+        insta::assert_json_snapshot!(data);
+    }
+
+    #[tokio::test]
+    async fn test_executor_arguments_default_value() {
+        //  NOTE: This test makes a real HTTP call
+        let request = Request::new("query {post {id title}}");
+        let executor = TestExecutor::try_new().await.unwrap();
+        let response = executor.run(request).await.unwrap();
+        let data = response.data;
+
+        insta::assert_json_snapshot!(data);
+    }
+
+    #[tokio::test]
+    async fn test_executor_variables() {
+        //  NOTE: This test makes a real HTTP call
+        let query = r#"
+            query user($id: Int!) {
+              user(id: $id) {
+                id
+                name
+              }
+            }
+        "#;
+        let request = Request::new(query);
+        let executor = TestExecutor::try_new().await.unwrap();
+
+        match executor.run(request).await {
+            Ok(_) => panic!("Should fail with unresolved variable"),
+            Err(err) => assert_eq!(
+                err.to_string(),
+                "ResolveInputError: Variable `id` is not defined"
+            ),
+        };
+
+        let request = Request::new(query);
+        let request = request.variables(Variables::from_iter([("id".into(), ConstValue::from(1))]));
```

Comment:
```suggestion
        let request = request.variables([("id".into(), ConstValue::from(1))]);
```

No need exposing `Variables` publicly just for tests.

Reply:
applied the suggestion